src/composer.ts

```diff
@@ -698,4 +736,46 @@ export class AtomicTransactionComposer {
       methodResults,
     };
   }
+
+  /**
+   * Parses a single ABI Method transaction log into a ABI result object.
+   *
+   * @param method
+   * @param methodResult
+   * @param pendingInfo
+   * @returns An ABIResult object
+   */
+  static parseMethodResponse(
+    method: ABIMethod,
+    methodResult: ABIResult,
+    pendingInfo: Record<string, any>
+  ): ABIResult {
+    const returnedResult: ABIResult = methodResult;
+    try {
+      returnedResult.txInfo = pendingInfo;
+      if (method.returns.type !== 'void') {
+        const logs: string[] = pendingInfo.logs || [];
+        if (logs.length === 0) {
+          throw new Error('App call transaction did not log a return value');
+        }
+
+        const lastLog = Buffer.from(logs[logs.length - 1], 'base64');
+        if (
+          lastLog.byteLength < 4 ||
+          !lastLog.slice(0, 4).equals(RETURN_PREFIX)
```

Comment:
why do we name return_prefix but assume its length is 4

Reply:
[ARC-04](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0004.md#standard-format) magic number - the [change predates this PR](https://github.com/algorand/js-algorand-sdk/pull/743/files#diff-534b9601ad1de4bb83d07dc0cb5af85424ccf602bcd8f4c0a2d3b79def06e8aeL677), but seems reasonable to specify a constant here or take a byteLength